client/v1/healthz_test.go

```diff
@@ -214,3 +309,156 @@ func TestBlockUntilServerReadyWithInvalidResponse(t *testing.T) {
 		t.Error("BlockUntilServerReady() with invalid response should return error")
 	}
 }
+
+func TestCheckHealthzConcurrent(t *testing.T) {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path != "/healthz" {
+			http.NotFound(w, r)
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+		json, _ := server.DefaultHealthz.JSON()
+		_, err := w.Write(json)
+		require.NoError(t, err)
+	}))
+	defer srv.Close()
+
+	var wg sync.WaitGroup
+	for i := 0; i < 10; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err := CheckHealthz(context.Background(), srv.URL)
+			assert.NoError(t, err)
+		}()
+	}
+	wg.Wait()
+}
+
+func TestBlockUntilServerReadyExactRetries(t *testing.T) {
+	retryCount := 0
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		retryCount++
+		// Only succeed after 29 retries (30th attempt)
+		if retryCount < 30 {
+			w.WriteHeader(http.StatusServiceUnavailable)
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+		json, _ := server.DefaultHealthz.JSON()
+		_, err := w.Write(json)
+		require.NoError(t, err)
+	}))
+	defer srv.Close()
+
+	ctx := context.Background()
+	err := BlockUntilServerReady(ctx, srv.URL, WithCheckInterval(1*time.Millisecond))
+	assert.NoError(t, err)
+	assert.Equal(t, 30, retryCount, "Expected exactly 30 retries")
+}
+
+func TestBlockUntilServerReadyWithSlowResponse(t *testing.T) {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(50 * time.Millisecond)
+		w.WriteHeader(http.StatusOK)
+		json, _ := server.DefaultHealthz.JSON()
+		_, err := w.Write(json)
+		require.NoError(t, err)
+	}))
+	defer srv.Close()
+
+	// Test with a client timeout shorter than server response time
+	client := &http.Client{Timeout: 20 * time.Millisecond}
+	err := BlockUntilServerReady(context.Background(), srv.URL,
+		WithHTTPClient(client),
+		WithCheckInterval(10*time.Millisecond))
+	assert.Error(t, err)
+	assert.Equal(t, err, ErrServerNotReady)
+
+	// Test with a client timeout longer than server response time
+	client = &http.Client{Timeout: 100 * time.Millisecond}
+	err = BlockUntilServerReady(context.Background(), srv.URL,
+		WithHTTPClient(client),
```

Comment:
The caller of BlockUntilServerReady doesn't pass in any client, do we really need this test? I thought for the function `BlockUntilServerReady` we only needed ctx, url, interval, and number of retries?